app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -164,10 +165,12 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         // Must be after we set the content view
         if (isVisuallyComplete) {
             rootContainer.doOnPreDraw {
+                val homeActivityWeakRef = WeakReference<HomeActivity>(this)
+
                 // This delay is temporary. We are delaying 5 seconds until the performance
                 // team can locate the real point of visual completeness.
                 it.postDelayed({
```

Comment:
I'm concerned that:
- lambdas (or any anonymous class, for that matter) always capture `this`
- someone will just add something to this call that creates a new memory leak (e.g. logging a member variable)

Can we encapsulate this code such that this memory leak cannot happen easily (e.g. make it non-anonymous)?

---

fwiw, I have felt it weird that we're doing this low-level stuff in a lifecycle method (one of those red flags I mentioned) – it feels like this whole bit should be in at least a helper function, if not an entirely separately file.

Reply:
has to add a static function inside `HomeActivity` since `RunWhenReadyQueue` class is read only.

Reply:
Sorry, I missed this comment before my review. It's defined in android-components – maybe we could wrap it or just define it in a new file?